yt/utilities/lib/element_mappings.pyx

```diff
@@ -256,7 +256,7 @@ cdef class P1Sampler3D(ElementSampler):
             col1[i] = vertices[3 + i]     - vertices[9 + i]
             col2[i] = vertices[6 + i]     - vertices[9 + i]
 
-        d = determinant_3x3(col0, col1, col2)
+        d = determinant_3x3(col0, col1, col2) + 1e-16
```

Comment:
In general I'm not opposed to having a specific offset specified, *but*, it may help if we either used a constant provided by numpy or math.h *or* if we were to use a `DEF` (if those are still allowed) to put it at the top-level.

Reply:
that makes sense to me!

Reply:
`misc_utilities.pyx` uses

```python
     cdef np.float64_t eps = np.finfo(np.float64).eps
```

in a couple spots, so I can try that here too. If I follow that example, I'd add it to the individual functions though -- do you think it's worth putting it at a higher level than function level? Might be able to add it as a attribute of the base `ElementSampler` class. Or maybe as a standalone variable in the `element_mappings.pxd` ?

Reply:
oh actually, I can add a 

```python
from libc.float cimport DBL_MIN
``` 
at the top and use that `DBL_MIN` throughout.

Reply:
I think we can consider ˋDEF` still allowed. I'll handle the migration to whatever ends up replacing it in Cython 3 !